From 568a508f1e07c1e46b8a47f8f25617e84439968f Mon Sep 17 00:00:00 2001
From: Sebastian Reichel <sebastian.reichel@collabora.com>
Date: Wed, 5 Nov 2025 18:44:09 +0100
Subject: [PATCH 104/108] arm64: dts: rockchip: Enable ALDPS on both network
 interfaces of RK3576 EVB1

ALDPS reduces the power consumption while the network interfaces
are enabled, but no cable is plugged in. The datasheet specifies:

> Whole system power consumption in ALDPS low power mode (with PLL
> turned off) is 10.3mW for the RTL8211F(I), and 23.1 mW for the
> RTL8211FD(I).

For the 3.3V line supplying the network PHYs I see the following
power numbers:

 * both interfaces down: 239mW
 * both interface up, 1 cable plugged: 1005mW
 * both interfaces up, 1 cable plugged after this series: 995mW

Signed-off-by: Sebastian Reichel <sebastian.reichel@collabora.com>
---
 arch/arm64/boot/dts/rockchip/rk3576-evb1-v10.dts | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/arch/arm64/boot/dts/rockchip/rk3576-evb1-v10.dts b/arch/arm64/boot/dts/rockchip/rk3576-evb1-v10.dts
index db8fef7a4..c5584c26d 100644
--- a/arch/arm64/boot/dts/rockchip/rk3576-evb1-v10.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3576-evb1-v10.dts
@@ -717,6 +717,7 @@ rgmii_phy0: ethernet-phy@1 {
 		assigned-clock-rates = <25000000>;
 		pinctrl-names = "default";
 		pinctrl-0 = <&rgmii_phy0_rst>;
+		realtek,aldps-enable;
 		reset-assert-us = <20000>;
 		reset-deassert-us = <100000>;
 		reset-gpios = <&gpio2 RK_PB5 GPIO_ACTIVE_LOW>;
@@ -732,6 +733,7 @@ rgmii_phy1: ethernet-phy@1 {
 		assigned-clock-rates = <25000000>;
 		pinctrl-names = "default";
 		pinctrl-0 = <&rgmii_phy1_rst>;
+		realtek,aldps-enable;
 		reset-assert-us = <20000>;
 		reset-deassert-us = <100000>;
 		reset-gpios = <&gpio3 RK_PA3 GPIO_ACTIVE_LOW>;
-- 
2.47.3

